pallets/payment-streams/src/lib.rs

```diff
@@ -801,36 +801,38 @@ pub mod pallet {
-        /// remaining payment streams to be able to recover its deposits.
+        /// Dispatchable extrinsic that allows a user flagged as without funds to pay the Providers that still have payment streams
+        /// with it, in order to recover as much of its deposits as possible.
         ///
         /// The dispatch origin for this call must be Signed.
         /// The origin must be the User that has been flagged as without funds.
         ///
         /// This extrinsic will perform the following checks and logic:
         /// 1. Check that the extrinsic was signed and get the signer.
         /// 2. Check that the user has been flagged as without funds.
-        /// 3. Release the user's funds that were held as a deposit for each payment stream.
-        /// 4. Get the payment streams of the user and charge them, paying the Providers for the services.
+        /// 3. Release the user's funds that were held as a deposit for each payment stream to be paid.
+        /// 4. Get the payment streams that the user has with the provided list of Providers, and pay them for the services.
         /// 5. Delete the charged payment streams of the user.
         ///
-        /// Emits a 'UserPaidAllDebts' event when successful.
+        /// Emits a 'UserPaidSomeDebts' event when successful if the user has remaining debts. If the user has successfully paid all its debts,
+        /// it emits a 'UserPaidAllDebts' event.
         ///
-        /// Notes: this extrinsic iterates over payment streams of the user and charges them, so it can be expensive in terms of weight.
-        /// The fee to execute it should be high enough to compensate for the weight of the extrinsic, without being too high that the user
-        /// finds more convenient to wait for Providers to get its deposits one by one instead.
+        /// Notes: this extrinsic iterates over the provided list of Providers, getting the payment streams they have with the user and charging
+        /// them, so the execution could get expensive. It's recommended to provide a list of Providers that the user actually has payment streams with,
+        /// which can be obtained by calling the `get_providers_with_payment_streams_with_user` runtime API.
+        /// There was an idea to limit the amount of Providers that can be received by this extrinsic using a constant in the configuration of this pallet,
+        /// but the correct benchmarking of this extrinsic should be enough to avoid any potential abuse.
         #[pallet::call_index(8)]
-        #[pallet::weight(T::WeightInfo::pay_outstanding_debt(*amount_of_streams_to_pay))]
+        #[pallet::weight(T::WeightInfo::pay_outstanding_debt(providers.len().try_into().unwrap_or(u32::MAX)))]
         pub fn pay_outstanding_debt(
             origin: OriginFor<T>,
-            amount_of_streams_to_pay: u32,
+            providers: Vec<ProviderIdFor<T>>,
```

Comment:
unbounded 😱😆

Reply:
It's bounded by the weight of the extrinsic! Basically the caller of this will pay for its execution proportional to the amount of Providers, so it shouldn't be an issue. I thought about making it a bounded vec and bounding it with a constant of the Config of the pallet, maybe that's better? @ffarall cc'ing you just in case

Reply:
Worst case scenario, the user sends a transaction that can take up A LOT of weight, but we know that in advance and will charge him for it. Whether or not this could result in another issue... unclear to me.